google-cloud-firestore/src/main/java/com/google/cloud/firestore/PipelineSource.java

```diff
@@ -0,0 +1,166 @@
+    this.rpcContext = rpcContext;
+  }
+
+  /**
+   * Creates a new {@link Pipeline} that operates on the specified Firestore collection.
+   *
+   * @param path The path to the Firestore collection (e.g., "users").
+   * @return A new {@code Pipeline} instance targeting the specified collection.
+   */
+  @Nonnull
+  @BetaApi
+  public Pipeline collection(@Nonnull String path) {
+    return collection(path, new CollectionOptions());
+  }
+
+  @Nonnull
+  @BetaApi
+  public Pipeline collection(@Nonnull String path, CollectionOptions options) {
+    return new Pipeline(this.rpcContext, new Collection(path, options));
+  }
+
+  @Nonnull
+  @BetaApi
+  public Pipeline collection(@Nonnull CollectionReference ref) {
+    if (!this.rpcContext.getFirestore().equals(ref.getFirestore())) {
+      throw new IllegalArgumentException(
+          "Invalid CollectionReference. The Firestore instance of the CollectionReference must"
+              + " match the Firestore instance of the PipelineSource.");
+    }
+
+    return collection(ref.getPath(), new CollectionOptions());
+  }
+
+  /**
+   * Creates a new {@link Pipeline} that operates on all documents in a collection group.
+   *
+   * <p>A collection group consists of all collections with the same ID. For example, if you have
+   * collections named "users" under different documents, you can query them together using a
+   * collection group query.
+   *
+   * @param collectionId The ID of the collection group.
+   * @return A new {@code Pipeline} instance targeting the specified collection group.
+   */
+  @Nonnull
+  @BetaApi
+  public Pipeline collectionGroup(@Nonnull String collectionId) {
+    return collectionGroup(collectionId, new CollectionGroupOptions());
+  }
+
+  @Nonnull
+  @BetaApi
+  public Pipeline collectionGroup(@Nonnull String collectionId, CollectionGroupOptions options) {
+    Preconditions.checkArgument(
+        !collectionId.contains("/"),
+        "Invalid collectionId '%s'. Collection IDs must not contain '/'.",
+        collectionId);
+    return new Pipeline(this.rpcContext, new CollectionGroup(collectionId, options));
+  }
+
+  /**
+   * Creates a new {@link Pipeline} that operates on all documents in the Firestore database.
+   *
+   * <p>Use this method with caution as it can lead to very large result sets. It is usually only
+   * useful at development stage.
+   *
+   * @return A new {@code Pipeline} instance targeting all documents in the database.
+   */
+  @Nonnull
+  @BetaApi
+  public Pipeline database() {
+    return new Pipeline(this.rpcContext, new Database());
+  }
+
+  /**
+   * Creates a new {@link Pipeline} that operates on a specific set of Firestore documents.
+   *
+   * @param docs The {@link DocumentReference} instances representing the documents to include in
+   *     the pipeline.
+   * @return A new {@code Pipeline} instance targeting the specified documents.
+   */
+  @Nonnull
+  @BetaApi
+  public Pipeline documents(DocumentReference... docs) {
```

Comment:
Android also has an overload taking string `fun documents(vararg documents: String): Pipeline` this can be added later